app/src/main/java/com/example/sharingang/DetailedItemFragment.kt

```diff
@@ -73,32 +82,84 @@ class DetailedItemFragment : Fragment() {
         return binding.root
     }
 
-    private fun initiateWishlistButton(){
-        if(currentUserProvider.getCurrentUserId() != null){
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.menu_detailed, menu)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+        val sell = menu.findItem(R.id.menuSell)
+        val resell = menu.findItem(R.id.menuResell)
+        if (!args.item.userId.equals(currentUserProvider.getCurrentUserId())) {
+            menu.findItem(R.id.menuEdit).isVisible = false
+            sell.isVisible = false
+            resell.isVisible = false
+        } else {
+            resell.isVisible = item?.sold ?: args.item.sold
+            sell.isVisible = !resell.isVisible
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.menuEdit -> {
+                view?.findNavController()?.navigate(
+                    DetailedItemFragmentDirections.actionDetailedItemFragmentToNewEditFragment(args.item)
+                )
+                true
+            }
+            R.id.menuSell, R.id.menuResell -> {
+                updateSold()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun updateSold() {
+        lifecycleScope.launch(Dispatchers.IO) {
+            itemViewModel.sellItem(item)
+            getItem()
+        }
+    }
+
+    private fun getItem() {
+        Log.d("SELL_STATUS", "nice")
```

Comment:
Maybe loadItem would be a better name, getItem looks like a getter that should return the item.
Also there is a nice debug log 😆

Reply:
Whoops, my bad 😆